workflow/scripts/plot_diffexp_heatmap.R

```diff
@@ -24,47 +42,31 @@ if (snakemake@params[["predef_genelist"]]$activate == TRUE) {
     )
     selectedgenes$gene <- NULL
     selectedgenes$transcript <- NULL
-    if (all(selectedgenes == 0)) {
-        txt <- "cannot plot, all values are zero"
-        pdf(snakemake@output[["diffexp_heatmap"]], height = 10, width = 10)
-        plot.new()
-        text(.5, .5, txt, font = 2, cex = 1.5)
-        dev.off()
-    # If number of transcripts are more than 100,
-    # Since for RNA-seq multiple transcripts may present for single gene.
-    } else if (nrow(selectedgenes) > 100) {
-        pdf_height <- round(nrow(selectedgenes) / 7)
-        pdf(snakemake@output[["diffexp_heatmap"]],
-            height = pdf_height, width = ncol(selectedgenes)
-        )
-        pheatmap(selectedgenes, selectedgenes,
-            cluster_rows = FALSE,  display_numbers = TRUE,
-                cellheight = 10, scale = "row"
-        )
-        dev.off()
-    } else {
-        fontsize_row <- 10 - nrow(selectedgenes) / 15
-        pdf(snakemake@output[["diffexp_heatmap"]])
-        pheatmap(selectedgenes, selectedgenes,
-            cluster_rows = FALSE, fontsize_row = fontsize_row, scale = "row"
-        )
-        dev.off()
-    }
+}
+# Checks if selectedgenes variable contain genes that have expression values
+if (all(selectedgenes == 0)) {
+    txt <- "cannot plot, all values are zero"
+    pdf(snakemake@output[["diffexp_heatmap"]], height = 10, width = 10)
+    plot.new()
+    text(.5, .5, txt, font = 2, cex = 1.5)
+    dev.off()
+    # If number of transcripts is more than 100,
+    # Since for RNA-seq multiple transcripts may present for a single gene.
+} else if (nrow(selectedgenes) > 100) {
+    pdf_height <- round(nrow(selectedgenes) / 7)
+    pdf(snakemake@output[["diffexp_heatmap"]],
+        height = pdf_height, width = ncol(selectedgenes)
+    )
+    pheatmap(selectedgenes, selectedgenes,
+        cluster_rows = FALSE, display_numbers = TRUE,
+        cellheight = 10, scale = "row"
+    )
+    dev.off()
 } else {
-    # Adding gene name to corresponding transcript id from sleuth file
-    rownames(sleuth_file) <- paste(
-        sleuth_file$transcript,
-        ":", sleuth_file$gene
+    pdf(
+        file = snakemake@output[["diffexp_heatmap"]],
+        height = 10, width = 10
```

Comment:
Why having this hardcoded? Does the number of cols not play a role here like above with nrow > 100? Why does above case not always work?

Reply:
As I understood the pdf function does not dynamically adjust the height and width. For the above case, we are selecting the top 50 variable genes, which by default remain the same (number of cols). The above case ( nrow > 100) also works here and I have updated the code.